py/desispec/fluxcalibration.py

```diff
@@ -712,8 +717,14 @@ def match_templates(wave, flux, ivar, resolution_data, stdwave, stdflux, teff, l
                    "template_id":template_id}
         func_args.append( arguments )
 
-
-    if ncpu > 1:
+    if comm is not None and comm.Get_size() > 1: # MPI mode & more than one rank per star
+        delta, remainder = len(func_args) // size, len(func_args) % size
+        start = rank * delta + min(rank, remainder) # Start of the args to process
+        end = (rank + 1) * delta + min(rank + 1, remainder) # End of the args to process
+        results = list(map(_func, func_args[start:end]))
```

Comment:
This looks like it is dividing `func_args` into contiguous chunks for each mpi rank. Is that done to preserve the ordering of results? If there's a reason not to use the strided chunking pattern here (e.g. `func_args[rank::size]`), I suggest adding that reason in a comment.

Reply:
I agree the `func_args[rank::size]` is cleaner to mentally parse, and follows a common MPI striding pattern.  If we do need/want to maintain order (usually desireable all other things being equal), this might be simpler to maintain:
```
my_args = np.array_split(func_args, size)[rank]
results = list(map(_func, my_args))
```
i.e. `np.array_split` takes care of splitting the array into `size` different sub-arrays, while taking care of boundary conditions, off-by-one bugs, etc.

But: if you do go that route, test that it does what you intend; I haven't parsed your code in detail to make sure...

Reply:
I tried the unordered striding pattern and the results did not change, so for brevity I'll use `func_args[rank::size]`. @sbailey Thanks for mentioning `np.array_split`! I was not aware of this function before.